app/controllers/course/assessment/questions_controller.rb

```diff
@@ -2,8 +2,23 @@
 class Course::Assessment::QuestionsController < Course::Assessment::ComponentController
   before_action :authorize_assessment
 
+  # Use method to build new questions.
+  #
+  # Cancancan uses `assessment.specific_questions.build` to build a resource, which will break since the specific
+  # questions are nested through `question_assessments` and AR does not support build associations with nested
+  # has_many through.
+  def self.build_and_authorize_new_question(question_name, options)
```

Comment:
Can we write tests for this method, to ensure that the instance variables are set correctly?

Reply:
Actually our various feature specs ensure that the variable is set correctly already. Think I will add one more controller test for this.

Reply:
@weiqingtoh have added the tests in text response controller to check the instance variables.